### ISSUE TITLE
Eager glob expansion

### ISSUE THREAD

**Issue by Akkuma** [lukeed/taskr#178]
Currently the code calls `expand` aka `globby` each time `unwrap` is called. It seems like it would be beneficial to eagerly evaluate in `source`, so it is done once for each `source`, rather than multiple times per wrap/target.

**Comment by lukeed** [comment]
Yeah definitely, been thinking about this too. 
Would need a getter method that asks for a saved "stash", before unwrapping. 

Probably live at `this._.files`

**Comment by Akkuma** [comment]
I don't think you need a getter method. I think all you need is this:

``` js
Fly.prototype.source = function (globs) {
  globs = arrify(globs);
  var flattenedGlobs = flatten(globs);

  assign(this, {
    _: {
      filters: [],
      globs: flattenedGlobs,
      files: Promise.all(flattenedGlobs.map(expand))
    }
  });

  this._.cat = undefined
  _('source %o', this._.globs)
  return this
}
```

in unwrap you now use roughly

``` js
Fly.prototype.unwrap = function (onResolved, onRejected) {
    this._.files.then(onResolved).catch(onRejected);
}
```

**Comment by lukeed** [comment]
Yeah, true -- by the time any filter/plugin is called, `source()` has already been run... early morning haze 😴

**Comment by lukeed** [comment]
Closed by #189
